### PR TITLE
Support Webpack 5

### LINKED ISSUE #387: webpack5: Unable to view parsed and gzip stats for the main entrypoint
#### Issue description
After upgrading to webpack 5, I notice that the parsed and gzip stats do not show up for the main.[hash].js chunk. All the other chunks are fine.

I confirmed that running it via the cli results in: `Couldn't parse bundle asset ...`

#### Technical info

<pre>
  System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 1.16 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 13.11.0 - ~/.nvm/versions/node/v13.11.0/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.13.7 - ~/.nvm/versions/node/v13.11.0/bin/npm
  npmPackages:
    compression-webpack-plugin: ^4.0.0 => 4.0.0 
    html-webpack-harddisk-plugin: ^1.0.1 => 1.0.1 
    html-webpack-plugin: ^4.2.1 => 4.3.0 
    webpack: ^5.1.0 => 5.1.3 
    webpack-bundle-analyzer: ^3.8.0 => 3.9.0 
    webpack-cli: 4 => 4.0.0 
    webpack-dev-middleware: ^3.7.2 => 3.7.2 
    webpack-dev-server: ^3.11.0 => 3.11.0 
    webpack-hot-middleware: ^2.25.0 => 2.25.0 
    webpack-merge: ^4.2.2 => 4.2.2 
</pre>

#### Debug info
I normally use it as a plugin, i.e. `new BundleAnalyzerPlugin()`

[stats.json.zip](https://github.com/webpack-contrib/webpack-bundle-analyzer/files/5418105/stats.json.zip)

### LINKED ISSUE #388: webpack 5: Detailed bundle structure is gone
#### Issue description

There are no any details about packages in bundle report

#### Technical info
<pre>
 System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 1.97 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.1.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
  npmPackages:
    eslint-webpack-plugin: 2.1.0 => 2.1.0 
    html-webpack-plugin: 4.5.0 => 4.5.0 
    webpack: 5.2.0 => 5.2.0 
    webpack-bundle-analyzer: 3.9.0 => 3.9.0 
    webpack-cli: 3.3.12 => 3.3.12 
    webpack-dev-server: 3.11.0 => 3.11.0 
</pre>

#### Debug info

I used it as plugin in webpack config:

<pre>
new BundleAnalyzerPlugin({
      analyzerMode: 'server',
      analyzerHost: '127.0.0.1',
      analyzerPort: 12345,
      logLevel: 'info',
    })
</pre>

[stats.zip](https://github.com/webpack-contrib/webpack-bundle-analyzer/files/5424444/stats.zip)
Снимок экрана 2020-10-22 в 19 59 52

### PR DESCRIPTION
**Pull request by th0r**
- [x] Remove support for Node.js 6 and 8 (set minimum required version to 10.13.0)
- [x] Update dependencies
- [x] Support multiple Webpack versions in tests
- [x] Fix #387
- [x] Fix #388
- [x] Bump major version
- [x] Update changelog

### CODE REVIEW

**Comment by th0r** [comment]
@valscion do you want to take a look?

**Comment by valscion** [comment]
Yup, I'll check this out :)

File: package.json [hunk 1]
```diff
@@ -16,15 +16,16 @@
...
   "scripts": {
     "start": "gulp watch",
     "build": "gulp build",
     "npm-publish": "npm run lint && npm run build && npm test && npm publish",
     "lint": "eslint --ext js,jsx .",
-    "test": "mocha --exit --require @babel/register",
-    "test-dev": "mocha --watch --require @babel/register"
+    "install-test-webpack-versions": "./bin/install-test-webpack-versions.sh",
+    "test": "npm run install-test-webpack-versions && mocha --exit --require @babel/register",
+    "test-dev": "npm run install-test-webpack-versions && mocha --watch --require @babel/register"
```
Comment (valscion) [inline package.json:28]: The `mocha --watch` now is constantly re-running the tests when `npm run test-dev` is started.

It seems that this is caused by the support for testing different webpack versions and those tests outputting files to `test/output` directory, triggering mocha's watcher to rerun the tests as soon as they are finished.

I was able to leverage `--watch-ignore` flag as [described in the docs](https://mochajs.org/#-watch-w) to make this `test-dev` stop until changes are done to some tests ☺️ 

```suggestion
    "test-dev": "npm run install-test-webpack-versions && mocha --watch --watch-ignore test/output --require @babel/register"
```

File: src/analyzer.js [hunk 2]
```diff
@@ -92,21 +92,52 @@ function getViewerData(bundleStats, bundleDir, opts) {
...
+      for (const statModule of assetModules) {
+        if (parsedModules[statModule.id]) {
           statModule.parsedSrc = parsedModules[statModule.id];
+        } else if (isEntryModule(statModule)) {
+          unparsedEntryModules.push(statModule);
         }
-      });
+      }
+
+      // Webpack 5 changed bundle format and now entry modules are concatenated and located at the end on the it.
```
Comment (valscion) [inline src/analyzer.js:118]: Is here a typo? The "at the end on the it" confused me and I'm not sure what it refers to, so I'm afraid I'm not able to provide a code suggestion to fix it.

File: src/analyzer.js [hunk 3]
```diff
@@ -92,21 +92,52 @@ function getViewerData(bundleStats, bundleDir, opts) {
...
+      // Because of this they basically become a concatenated module, for which we can't even precisely determine its
+      // parsed source as it's located in the same scope as all Webpack runtime helpers.
+      if (unparsedEntryModules.length && assetSources) {
+        if (unparsedEntryModules.length === 1) {
+          // So if there is only one entry we consider its parsed source to be all the bundle code excluding code
+          // from parsed modules.
+          unparsedEntryModules[0].parsedSrc = assetSources.runtimeSrc;
+        } else {
+          // If there are multiple entry points we move all of them under synthetic concatenated module.
```
Comment (valscion) [inline src/analyzer.js:127]: Nice comments! They makes the conditionals easier to follow :+1:

**Review by valscion — CHANGES_REQUESTED** [review]
Nice, I'm thrilled with having the possibility of creating tests for multiple different webpack versions!

At first I got mysterious errors for various tests and was super confused — then I cloned the entire repository from a clean slate and discovered I had forgot to run `npm run build` before running tests 😅. After running that, all tests passed for me locally, too :tada:

My comments are only minor in nature. I didn't go through all of the changes you did to parsing bundles logic as I trust the tests to be good enough to ensure that the logic is correct :relaxed:

File: test/webpack-versions/4.44.2/package.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (th0r) [reply test/webpack-versions/4.44.2/package.json:1]: Yep, nice suggestion! Thanks

File: test/webpack-versions/5.4.0/package.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
```
[the remark this answers is not in the document]
Comment (th0r) [reply test/webpack-versions/5.4.0/package.json:1]: Fixed

File: package.json (hunk 1 above)
Comment (th0r) [reply package.json:28]: Nice catch! Fixed

File: src/analyzer.js (hunk 2 above)
Comment (th0r) [reply src/analyzer.js:118]: Hah, yep, it should be `and located at the end of it.`

**Review by valscion — APPROVED** [review]
Looks good to me! Great job! :+1: :+1: :clap: :100: :tada: :rocket:

**Pull request merged**